front/components/actions/retrieval/utils.ts

```diff
@@ -0,0 +1,87 @@
+import type { ConnectorProvider, RetrievalDocumentType } from "@dust-tt/types";
+
+type ConnectorProviderDocumentType =
+  | Exclude<ConnectorProvider, "webcrawler">
+  | "document";
+
+const providerMap: Record<string, ConnectorProviderDocumentType> = {
+  "managed-slack": "slack",
+  "managed-notion": "notion",
+  "managed-google_drive": "google_drive",
+  "managed-github": "github",
+  "managed-confluence": "confluence",
+  "managed-intercom": "intercom",
+};
+
+const providerRegex = new RegExp(`^(${Object.keys(providerMap).join("|")})`);
+
+function getProviderFromRetrievedDocument(
+  document: RetrievalDocumentType
+): ConnectorProviderDocumentType {
+  const match = document.dataSourceId.match(providerRegex);
+  if (match && match[1]) {
+    return providerMap[match[1]];
+  }
+
+  return "document";
+}
+
+function getTitleFromRetrievedDocument(
```

Comment:
Actually isn't the logic from this doc already defined in "@app/components/assistant/conversation/RetrievalAction"?

Reply:
Yeah I left a comment in the top of the legacy file. Today the logic is spread everywhere through the codebase. So the plan here is to first introduce new action details design before getting rid of all the legacy code.